### ISSUE TITLE
Implementation just for 2D point cloud?

### ISSUE THREAD

**Issue by HowcanoeWang** [Qutom/BoundaryDetection-Algorithm-for-Point-Cloud-Data#1]
Thanks a lot for implementing this algorithm in python for this paper!

When using this code, I got a problem.

According to the demo code in `main.py` and the figures in this project, did you convert the 3D point cloud to 2D point cloud first (by `plane_extractor.extract_plane(pc)`), and then run this algorithm on the produced 2D point cloud?

I found the output boundary was only 2D coordiante `(x, y, 0)`, and fit well with the 2D boundary detection

![image](https://user-images.githubusercontent.com/19968935/227753661-9397db14-3694-45b9-b694-3eebcbb68d01.png)

When using the latter code in `main.py` to convert back to 3D (I guess it provides such function for that), the result does not work well on the demo data by paper at all:

![image](https://user-images.githubusercontent.com/19968935/227753745-fcf3ab3b-0ae6-4aec-97bb-5e302e3de60d.png)

**Comment by Qutom** [comment]
Yes, I edited code to find the wall connected to the ceiling of the building. I removed z-axis info in `plane_extractor.extract_plane(pc)`. so It will work just for 2d point cloud

(++ circle class also supports only two dimensions. So expand to 3D you need modify  `Circle.py`, `BPD.py` and don't use `plane_extractor.extract_plane(pc)`)
